enterprise_gateway/services/external/k8s_client_factory.py

```diff
@@ -0,0 +1,31 @@
+"""Contains factory to create kubernetes api client instances using a single confguration"""
+import os
+
+from kubernetes import client, config
+from traitlets.config import SingletonConfigurable
+
+
+class KubernetesClientFactory(SingletonConfigurable):
+    """Manages kubernetes client creation from environment variables"""
+    def __init__(self) -> None:
+        """Maintain a single configuration object and populate based on environment"""
+        super().__init__()
+
+    def get_kubernetes_client(self, get_remote_if_available=True) -> client.ApiClient:
+        """Get kubernetes api client with appropriate configuration"""
+        kubernetes_config: client.Configuration = client.Configuration()
+        if os.getenv("KUBERNETES_SERVICE_HOST"):
+            # Running inside cluster
+            if os.getenv('EG_USE_REMOTE_CLUSTER') and get_remote_if_available:
```

Comment:
I am not sure this works as expected. This will evaluate to true if `EG_USE_REMOTE_CLUSTER` is set to _anything_ (including `EG_USE_REMOTE_CLUSTER=False`). 

We should confirm this has the same semantics as other env options defined in JEG. Do they look for True/False or just Set/Unset. If True/False, we should add a more rigorous check. If set/unset, we should use `EG_USE_REMOTE_CLUSTER=1` in the helm chart to denote this.